### PR TITLE
Companion for paritytech/substrate#12795

### PR DESCRIPTION
**Pull request by m-cat**
# PULL REQUEST

## Overview

This PR removes `parity-util-mem`. It upgrades some dependencies to fully remove it from `Cargo.lock`.

## Companion

Companion for paritytech/substrate#12795

## Related issue

https://github.com/paritytech/substrate/issues/12658

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2107206

**Review by acatangiu — APPROVED** [review]

**Comment by m-cat** [comment]
@ordian Does this need a cumulus companion? There are no code changes required in cumulus and the dependent job in CI succeeded. But would cumulus get updated if we do `bot merge` here without a companion? Also, when I do `diener` locally in cumulus it doesn't remove all instances of `parity-util-mem` from `Cargo.lock`. I still need to do `cargo upgrade -p memory-db -p kvdb -p kvdb-rocksdb -p kvdb-memorydb -p trie-bench --incompatible` after. Should that be in a separate non-companion PR?

File: node/overseer/src/memory_stats.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+use tikv_jemalloc_ctl::{epoch, stats, Error};
```
Comment (ordian) [inline node/overseer/src/memory_stats.rs:1]: nit: missing license header
https://github.com/paritytech/polkadot/blob/master/file_header.txt

File: src/main.rs [hunk 2]
```diff
@@ -20,6 +20,10 @@
 
 use color_eyre::eyre;
 
+/// Global allocator
+#[global_allocator]
+pub static ALLOC: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
```
Comment (ordian) [inline src/main.rs:25]: maybe we should add a note, that changing to another allocator it will require changing `memory_stats::MemoryAllocationTracker`

**Review by ordian — APPROVED** [review]
Nice!

**Comment by ordian** [comment]
> Should that be in a separate non-companion PR?

If it's not possible to fit into a companion, yes.

**Comment by ordian** [comment]
bot rebase

**Comment by ordian** [comment]
bot rebase

**Comment by ordian** [comment]
> @ordian Does this need a cumulus companion? There are no code changes required in cumulus and the dependent job in CI succeeded. But would cumulus get updated if we do `bot merge` here without a companion? Also, when I do `diener` locally in cumulus it doesn't remove all instances of `parity-util-mem` from `Cargo.lock`. I still need to do `cargo upgrade -p memory-db -p kvdb -p kvdb-rocksdb -p kvdb-memorydb -p trie-bench --incompatible` after. Should that be in a separate non-companion PR?

@m-cat could you submit a cumulus PR?

**Comment by m-cat** [comment]
Finally! Yes @ordian, will do!

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-36/1529/1

**Pull request merged**